### PR TITLE
Handle reserved words gracefully.

### LINKED ISSUE #270: Support variables named like reserved keywords
I encountered this when trying to use Clutz on _ng.material.components.switch_. The *.d.ts produced by Clutz contains:

```
declare namespace ಠ_ಠ.clutz.ng.material.components {
  var switch : any;
}
```

And then the Typescript compiler throws

> java/com/google/developers/console/web/crash/common/_common_ts_js_typings.d.ts:155302:7: Variable declaration expected.

### PR DESCRIPTION
**Pull request by rkirov**
Closes #270

### CODE REVIEW

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -99,6 +99,19 @@ public boolean apply(@Nullable ObjectType input) {
...
+   * Thanks to https://mathiasbynens.be/notes/reserved-keywords.
+   */
+  private static final Set<String> reservedJsWords = ImmutableSet.of(
+      "do", "if", "in", "for", "let", "new", "try", "var", "case", "else",
+      "enum", "eval", "null", "this", "true", "void", "with", "await", "break",
+      "catch", "class", "const", "false", "super", "throw", "while", "yield",
+      "delete", "export", "import", "public", "return", "static", "switch",
+      "typeof", "default", "extends", "finally", "package", "private",
+      "continue", "debugger", "function", "arguments", "interface",
+      "protected", "implements", "instanceof");
```
Comment (martine) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Alphabetize?  ¯\_(ツ)_/¯

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 2]
```diff
@@ -1107,7 +1190,7 @@ private void visitTypeAsPrimary(JSType type) {
       }
     }
 
-    private void visitType(JSType typeToVisit) {
+    void visitType(JSType typeToVisit) {
```
Comment (martine) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Is this necessary?

**Comment by martine** [comment]
As discussed offline, LGTM.

(We don't always want to do this transformation because namespaces are open, and we can't use assignments to build up the namespaces out of objects because this is d.ts file.)

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 2 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: nope, it's a inner class, so I can access all methods. forgot my java (never knew it anyways).

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 1 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: sure. done.

**Pull request merged**